builtin/providers/cloudstack/resource_cloudstack_security_group.go

```diff
@@ -0,0 +1,210 @@
+package cloudstack
+
+import (
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/xanzy/go-cloudstack/cloudstack"
+)
+
+func resourceCloudStackSecurityGroup() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceCloudStackSecurityGroupCreate,
+		Read:   resourceCloudStackSecurityGroupRead,
+		Delete: resourceCloudStackSecurityGroupDelete,
+
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"description": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+
+			"project": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+
+			"ingress_rules": &schema.Schema{
+				Type:     schema.TypeList,
+				ForceNew: true,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"cidr": &schema.Schema{
```

Comment:
`cidr` should be called `cidr_list` to stay consistent throughout the CloudStack provider.